src.ts/account.ts

```diff
@@ -173,8 +173,12 @@ export class Account {
         return this.signAndSendTransaction(this.accountId, [deleteKey(PublicKey.from(publicKey))]);
     }
 
-    async stake(publicKey: string | PublicKey, amount: BN): Promise<FinalExecutionOutcome> {
-        return this.signAndSendTransaction(this.accountId, [stake(amount, PublicKey.from(publicKey))]);
+    async stake(publicKey: string | BlsPublicKey, amount: BN): Promise<FinalExecutionOutcome> {
+        const status = await this.connection.provider.status();
+        if (status.version.version < '0.4.0') {
```

Comment:
this isn't very robust way to compare versions, makes sense to use something like https://github.com/omichelsen/compare-versions